webpack/assets/javascripts/react_app/components/ForemanModal/helpers.js

```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import ForemanModal from '.';
+
+export function extractModalNodes(children) {
+  children = React.Children.toArray(children);
+  // Extract header and footer from children, if provided
```

Comment:
Since it's a helper, IMO it makes more sense to write esdoc comments on the function.

Reply:
Just checked out esdoc for the first time.  I'm not sure what type I would write for the `children` param, since React children can be any type.

Reply:
I assume `PropTypes.node` should be fine.